Add documentation for check and fix: Small gaps, Fill gaps (#10326)
qgis/QGIS-Documentation: docs/user_manual/processing_algs/qgis/fixgeometry.rst

  :end-before: **end_algorithm_code_section**

.. _qgisfixgeometrygap:

Fill gaps
--------------------

Fills the gaps based on a gap and neighbors layer from the check algorithm.
Three different fixing methods are available, which will give different results.

.. figure:: img/fix_geometry_gap.png
   :align: center

   Before and after fixing gaps in polygon features using create new feature method.

.. seealso:: :ref:`qgischeckgeometrygap`

Parameters
..........

Basic parameters
^^^^^^^^^^^^^^^^

.. list-table::
   :header-rows: 1
   :widths: 20 20 20 40
   :class: longtable

   * - Label
     - Name
     - Type
     - Description
   * - **Input layer**
     - ``INPUT``
     - [vector: polygon]
     - Layer with the geometries to fix, same as input in the check algorithm.
   * - **Neighbors layer**
     - ``NEIGHBORS``
     - [vector: geometry]
     - Layer with the errors location, generated by the check algorithm.
   * - **Gaps layer**
     - ``GAPS``
     - [vector: polygon]
     - Polygon layer with the gaps geometries, generated by the check algorithm.
   * - **Method**
     - ``METHOD``
     - [enumeration]

     - Method to apply to the gap fixing. Possible values are:

>        * 0 -- ``Add to longest shared edge``:

Reviewer: From memory, this used to be (but I don't know if it changes anything)
```suggestion
       * 0 --- ``Add to longest shared edge``:
```

         adds the gap to the longest shared edge with neighboring features.
       * 1 -- ``Create new feature``:
         creates a new feature for each gap.
       * 2 -- ``Add to largest neighbouring area``:
         adds the gap to the neighboring feature with the largest area.
   * - **Field of original feature unique identifier**
     - ``UNIQUE_ID``
     - [tablefield: any]
     - Field storing unique values for feature identification in input layer,
       as set in the check algorithm.
   * - **Field of error id**
     - ``ERROR_ID_IDX``

>      - [tablefield: any]

Reviewer: I wonder if we shouldn't add these default. Particularly if we refer to the check alg output
```suggestion
     - [tablefield: any]

       Default: gc_errorid
```

  :end-before: **end_algorithm_code_section**

.. _qgisfixgeometrygap:

Fill gaps
--------------------

Fills the gaps based on a gap and neighbors layer from the :ref:`qgischeckgeometrygap` algorithm.
Three different fixing methods are available.

.. figure:: img/fix_geometry_gap.png
   :align: center

>    Before and after fixing gaps in polygon features using add to longest shared edge method.

Reviewer: Is it not instead?
```suggestion
   Before and after fixing gaps in polygon features using ``Add to largest neighbouring area`` method.
```
